Component/filter

catherinehoward opened the pull request:
Closes #87 

This pull request adds a new filter component.

It's not finished yet. Have a couple of queries:

1. Should the label propType be changed to optional? Not all the select elements have labels in Figma,
2. The mobile layout is almost there. Just need to tweak the spacing. Trying to get the label to display on the left for the tablet and Desktop layouts. Does the media query for this need to go in the Select component?

Any thoughts?

Commit:
initial files for Filter component

Commit:
updated styles for Filter Component

Commit:
added responsive styling for Filter component

Commit:
Add styles for filter component

Commit:
Updated Select component to include prop for displaying label on left

RobTF9 commented:
1. They should always have a label, if it's not in the design that's an oversight. Inputs/selects should always have a label.
2. It would depend on whether the Select needs this style independently of this component. So if you're going to use it elsewhere and you'll need to use this media query I'd add it to the select component itself.

Commit:
Added filter text and updated labels

Commit:
removed leftlabel prop from select component

Commit:
Added additional tests for Filter component

admtomas committed:
Code changed to eleminate styling errors

admtomas committed:
changed max width to match design

admtomas committed:
code cleanup, redundand code media query for tablet will work same for desktop

Commit:
Changes to Filter component and styling following review

Commit:
Updated test cases

Commit:
fix merge conflicts

Commit:
Added label text for select elements and small changes to button styling

catherinehoward commented:
I've included your changes @admtomas apart from the following:
1. The Filter.styles.js file:
Amended FilterText to padding: var(--smallest-space); which is the same as 

padding: var(--smallest-space) var(--smallest-space);

2. I originally included different widths and heights for the FilterWrapper for Desktop and Mobile layouts to match the design.
I've left them out for now if you think they should all be consistent.

3. For the ButtonContainer I've removed:

> button {
        padding: var(--small-space);
      }
	  
Think it looks better to have the Clear filters button directly underneath the search button with no padding
as per Figma design.

I had some merge conflicts because we were both working on the same branch. These are now fixed and my working tree is clean. Not sure why it's still saying it has 'conflicting files'.

DaveDangereux commented on src/components/Filter/Filter.stories.jsx:
```diff
@@ -0,0 +1,14 @@
+};
+
+export const TestFilter = () => (
+  <BrowserRouter>
+    <Filter />
+  </BrowserRouter>
```
Would you clarify the purpose of wrapping the filter in the router here? It's academic - it's only Storybook, after all. But I am curious.

admtomas replied:
Well when we do something again  and again then we just put some unnecessary code into next work:D
Its not important to have router here as we use router inside proper web page. but it doesnt damage the result.

catherinehoward replied:
I copied the code from somewhere else but didn't quite understand why it was there. Should've queried it!

DaveDangereux reviewed (approved):
Looks grand.

admtomas commented:
Filter component is behind main branch that is why it shows merging conflicts.
There are some files missing. Need to create new branch  named Filters and copied Filter files to it.

Please delete this pull request and branch and refer to new branch.

DaveDangereux commented:
Resolving merge conflicts is good practice, no?

Ah well, there'll be plenty more opportunities, no doubt.

catherinehoward commented:
Yes it is better to resolve conflicts where possible.
